numpy/core/src/multiarray/ctors.c

```diff
@@ -2187,7 +2187,11 @@ PyArray_FromInterface(PyObject *origin)
     PyArrayObject *ret;
     PyArray_Descr *dtype = NULL;
     char *data = NULL;
+#if defined(NPY_PY3K)
+    Py_buffer view;
```

Comment:
Same remark about scoping this to where it's used, to remove an `#if defined`

Reply:
I considered that, but some of the places would require free floating blocks, which I really don't like.  I don't see much gain here, especially as this will all get cleaned up later when 2.7 is dropped.

Reply:
Note that we will also be moving to C99 at some point after Python 2.7 and 3.4 are dropped, and it allows declarations after statements. Although, in cases like this, that probably just makes the code more difficult to understand as one never knows where to find the declaration...

Reply:
Some advantages of floating blocks:
* They rule out places where the variable can be used
* When we switch to C99, the braces around them can be removed
* They reduce stack usage, as the variable does not consume the stack for the whole function

Reply:
Still don't like floating blocks, haven't since Algol-68. If you really need them, you probably need a subroutine.

Reply:
Fair enough